File: puzzles/python3/power-of-thor1/power_of_thor1.py
from typing import Tuple


def solve(light_x: int, light_y: int, thor_x: int, thor_y: int) -> Tuple[str, int, int]:
    direction = ""

    if thor_y > light_y:
        direction += "N"
        thor_y -= 1
    elif thor_y < light_y:
        direction += "S"
        thor_y += 1

    if thor_x > light_x:
        direction += "W"
        thor_x -= 1
    elif thor_x < light_x:
        direction += "E"
        thor_x += 1

    return direction, thor_x, thor_y


if __name__ == "__main__":
    light_x, light_y, thor_x, thor_y = map(int, input().split())

    # game loop
    while True:
        direction, thor_x, thor_y = solve(light_x, light_y, thor_x, thor_y)
        print(direction)

File: puzzles/cpp/power-of-thor1.cpp
#include <iostream>

using namespace std;

int main() {
  int lightX; // the X position of the light of power
  int lightY; // the Y position of the light of power
  int thorX; // Thor's starting X position
  int thorY; // Thor's starting Y position
  cin >> lightX >> lightY >> thorX >> thorY; cin.ignore();

  // game loop
  while (true) {
    string direction = "";
    int remainingTurns; // The remaining amount of turns Thor can move. Do not remove this line.
    cin >> remainingTurns; cin.ignore();

    if (thorY > lightY) {
      direction += "N";
      thorY--;
    } else if (thorY < lightY) {
      direction += "S";
      thorY++;
    }

    if (thorX > lightX) {
      direction += "W";
      thorX--;
    } else if (thorX < lightX) {
      direction += "E";
      thorX++;
    }

    // A single line providing the move to be made: N NE E SE S SW W or NW
    cout << direction << endl;
  }
}
